fvm/environment/program_recovery.go

```diff
@@ -29,16 +29,23 @@ func RecoverProgram(
 	sc := systemcontracts.SystemContractsForChain(chainID)
 
 	fungibleTokenAddress := common.Address(sc.FungibleToken.Address)
+	nonFungibleTokenAddress := common.Address(sc.NonFungibleToken.Address)
 
-	if !isFungibleTokenContract(program, fungibleTokenAddress) {
-		return nil, nil
-	}
+	var code string
 
-	contractName := addressLocation.Name
+	switch {
+	case isFungibleTokenContract(program, fungibleTokenAddress):
+		code = RecoveredFungibleTokenCode(fungibleTokenAddress, addressLocation.Name)
 
-	code := RecoveredFungibleTokenCode(fungibleTokenAddress, contractName)
+	case isNonFungibleTokenContract(program, nonFungibleTokenAddress):
+		code = RecoveredNonFungibleTokenCode(nonFungibleTokenAddress, addressLocation.Name)
+	}
+
+	if code != "" {
+		return parser.ParseProgram(memoryGauge, []byte(code), parser.Config{})
+	}
```

Comment:
Can return on `default` case
```suggestion
	default:
		return nil, nil
	}

	return parser.ParseProgram(memoryGauge, []byte(code), parser.Config{})
```